src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -812,7 +812,8 @@ static int next_beyond_location(sd_journal *j, JournalFile *f, direction_t direc
         assert(j);
         assert(f);
 
-        (void) journal_file_read_tail_timestamp(j, f);
+        if (!FLAGS_SET(j->flags, SD_JOURNAL_READ_TAIL_TIMESTAMP_ONCE))
+                (void) journal_file_read_tail_timestamp(j, f);
```

Comment:
i'd really prefer if this conditioning would be hidden in some helper around journal_file_read_tail_timestamp(). i.e. define journal_file_read_tail_timestamp_maybe() or so, that just checks the flag, and if appropriate rereads the field or not.

And then switch the relevant uses of the function over.

Reply:
There's one case in add_any_file() where we probably should also use the _maybe() version of the call

Reply:
Instead of introducing _maybe() version, Now the flag is always checked in journal_file_read_tail_timestamp().